src/db/clickhouse.ts

```diff
@@ -1,26 +1,25 @@
-import ClickHouse = require('@apla/clickhouse');
+import { ClickHouseClient, createClient, Row } from '@clickhouse/client';
 import getConfig from '../config';
 
-let _client: any;
+let _client: ClickHouseClient;
 
 async function getClient() {
-  if (!_client) _client = new ClickHouse((await getConfig()).db.clickhouse);
+  if (!_client) _client = createClient((await getConfig()).db.clickhouse);
   return _client;
 }
 
-export async function query<T>(q: string): Promise<T[]> {
+export async function query<T>(q: string, options: any = {}): Promise<T[]> {
     const result: T[] = [];
-    await queryStream(q, row => result.push(row));
+    await queryStream(q, row => result.push(row), options);
     return result;
 }
 
-export async function queryStream<T = any>(q: string, onRow: (row: T) => void): Promise<void> {
+export async function queryStream<T = any>(q: string, onRow: (row: T) => void, options: any = {}): Promise<void> {
   return new Promise(async resolve => {
-    const stream = (await getClient()).query(q);
-    stream.on('data', (row: T) => onRow(row));
-    stream.on('end', () => {
-      resolve();
-    });
+    const resultSet = await (await getClient()).query({ query: q, format: 'JSONCompactEachRow', ...options });
```

Comment:
So in python kernel, what we need to change is table name ,db engine and `FORMAT JSONCompact` like this format, is this right?

Reply:
Nope, this is JavaScript kernel with JavaScript client, for Python kernel, you guys need to test it, if current Python kernel client is compatible with the new ClickHouse server version, you don't need to change anything. You only need to change Python code if the client is not compatible with new server.

Keep in mind that: JavaScript/TypeScript kernel code and Python kernel code are totally independent.

Reply:
But for sure, you need to change the database and table name for queries.

Reply:
OK